### ISSUE TITLE
undefined method `get_code' for Pocket:Module

### ISSUE THREAD

**Issue by a-chris** [andyw8/pocket-ruby#91]
Hi, I want to use this gem so first of all I tried running the **demo-server** but I always receive this error:
```
NoMethodError - undefined method `get_code' for Pocket:Module:
	demo-server.rb:34:in `block in <main>'
```

when the execution reaches the line
```
session[:code] = Pocket.get_code(redirect_uri: CALLBACK_URL)
```

I tried both installing the gem with bundler and cloning the repository to run the exact **demo-server.rb** included in this repository but nothing changes.

My computers: 
- Macbook M1 with Apple Silicon and Ruby 2.7.2/3.0.0
- Ubuntu 21.04, x64 and Ruby 2.7.2/3.0.0

**Comment by andyw8** [comment]
Hi @a-chris 

I think this may have been due to a recent refactoring – could you try with v0.3.0 of the gem?

**Comment by a-chris** [comment]
@andyw8 thx for the response, I tried the 0.3.0 version and I can confirm that it works.

I compared the 0.3.0 and the 0.4.0 source codes but I could not find anything wrong, any idea o f what could be wrong with the 0.4.0 version?

**Comment by a-chris** [comment]
I found out that in the 0.3.0 version Pocket module used to delegate all the OAuth methods to the client instance thanks to these methods override
```
# Delegate to Pocket::Client
  def self.method_missing(method, *args, &block)
    return super unless client.respond_to?(method)
    client.send(method, *args, &block)
  end

  # Delegate to Pocket::Client
  def self.respond_to_missing?(method)
    client.respond_to?(method) || super
  end
```
that have been deleted in the 0.4.0 version so I think you should revert this specific change to fix the issue or you can replace `Pocket.get_code` with `Pocket.client.get_code`, same for `authorize_url`

**Comment by andyw8** [comment]
Fixed in v0.5.0
